backend/src/main/java/de/codexbella/user/MongoUserDetailsService.java

```diff
@@ -0,0 +1,23 @@
+package de.codexbella.user;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class MongoUserDetailsService implements UserDetailsService {
+   private final UserService userService;
+
+   @Override
+   public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+      return userService.findByUsername(username)
+            .map(userDocument -> new User(userDocument.getUsername(), userDocument.getPassword(), List.of(new SimpleGrantedAuthority("ROLE_"+userDocument.getRole()))))
```

Comment:
You never use the role in your `SecurityConfig`, so you can delete the role code here and in all other places (`UserData, JwtAuthFilter`).